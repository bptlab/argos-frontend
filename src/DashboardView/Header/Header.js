import React, { Component } from 'react';
import './Header.css';
import Boiler from '../../assets/icons/boiler.svg'

class Header extends Component {
    render() {
        return (
            <header>
                <div className="container">
                    <div className="row d-flex align-items-center">
                        <div className="col-1 d-flex justify-content-around">
                            <a href="/">
                                <img className="product-icon" src={Boiler}/>
                            </a>
                        </div>
                        <div className="col-10 d-flex justify-content-around">
                            <h1>Early Warning Dashboard</h1>
                        </div>
                    </div>
                </div>
            </header>
        );
    }
}
export default Header;