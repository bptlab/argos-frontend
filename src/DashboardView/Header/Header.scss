header {
	.product-icon {
		height: 50px;
		@include media-breakpoint-up(uhd) {
			height: 120px;
		}
	}
	
	.settings-icon {
		color: $COLOR-font-light;
		font-size: 30px;
		cursor: pointer;
	}
}
