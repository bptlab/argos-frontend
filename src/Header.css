.whiteIcon > div > svg {
	color: white !important;
}