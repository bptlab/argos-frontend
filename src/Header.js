import React, { Component } from 'react';
import AppBar from 'material-ui/AppBar';

class Header extends Component {
	render() {
		return (
			<AppBar
				title="Argos"
				iconClassNameRight="muidocs-icon-navigation-expand-more"
			/>
		);
	}
}

export default Header;