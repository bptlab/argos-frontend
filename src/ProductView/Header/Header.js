import React, { Component } from 'react';
import StatusPoint from './StatusPoint/StatusPoint.js';
import ConfigurationHeader from './ConfigurationHeader/ConfigurationHeader.js';

class Header extends Component {
    render() {
        let test = {
            "1.0":[ "4.0.0", "4.0.1", "4.0.2" ],
            "1.1":[ "4.1.0", "4.1.1", "4.1.2" ],
            "1.2":[ "4.2.0", "4.2.1", "4.2.2" ],
            "2.0":[ "5.0.0", "5.0.1", "5.0.2" ]
        };
        return (
            <header>
                <div className="container">
                    <div className="row d-flex align-items-center">
                        <div className="hidden-xs-down col-sm-1 d-flex justify-content-center">
                            <a href="/#">
                                <i className="home-button fa fa-home" />
                            </a>
                        </div>
                        <div className="row col-xs-12 col-sm-10 d-flex justify-content-around">
                            <div className="col-12 d-flex justify-content-center">
                                <h1 className="dashboard-title">{this.props.product.name}</h1>
                            </div>
                            <div className="col-12 d-flex justify-content-center">
                                <ConfigurationHeader configurations={this.props.configurations} onChangeProductConfiguration={this.props.onChangeProductConfiguration}/>
                            </div>
                        </div>
                        <div className="hidden-xs-down col-sm-1 d-flex justify-content-around">
                            <StatusPoint product={this.props.product}/>
                        </div>
                    </div>
                </div>
            </header>
        );
    }
}

export default Header;
