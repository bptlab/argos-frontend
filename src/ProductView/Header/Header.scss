@import "../../definitions.scss";
header {
	background-color: $light-grey;
	-webkit-box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
	-moz-box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
	box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
	> * {
		padding: 10px 0px;
	}
	.home-button {
		font-size: 30px;
	}
}