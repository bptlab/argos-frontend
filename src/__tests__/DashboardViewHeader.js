import React from 'react';
import Header from '../DashboardView/Header/Header.js';
import renderer from 'react-test-renderer';

test('Render DashboardView Header', () => {
    const component = renderer.create(
        <Header />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});