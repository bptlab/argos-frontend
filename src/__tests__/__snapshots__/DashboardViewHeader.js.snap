exports[`test Render DashboardView Header 1`] = `
<header>
  <div
    className="container">
    <div
      className="row d-flex align-items-center">
      <div
        className="col-1 d-flex justify-content-around">
        <a
          href="/">
          <img
            alt="Overview page"
            className="product-icon"
            src="boiler.svg" />
        </a>
      </div>
      <div
        className="col-10 d-flex justify-content-around">
        <h1>
          Early Warning Dashboard
        </h1>
      </div>
    </div>
  </div>
</header>
`;
